### PR TITLE
Enchance Laravel Automations

### PR DESCRIPTION
**Pull request by moisish**
Update the Laravel Automations for

- Route Caching
- Config Caching
- Event Discovery
- Optimizing/Cache Views

### CODE REVIEW

**Comment by alexjustesen** [comment]
2c on this PR, it would probably be better to run `artisan view:clear` to clear any cached views and then `artisan optimize` which does the route and cache config for you.

**Comment by moisish** [comment]
Why do you want to clear the view cache? Also running them individual is more flexible since you can disable one

**Comment by gtjamesa** [comment]
I've just discovered this repo, and after reading through the docs, I thought this would be a great addition. Luckily, someone had already submitted a PR 😁 I think configuring each cache is the better option, particularly as I have a legacy project that cannot cache its routes.

**Comment by dakira** [comment]
I don't see why this would be a good idea. There is no benefit in re-doing the caching every time the image starts. The code doesn't change unless the image content changes. A better place (IMHO) would be to do the caching in your build process, right?

**Comment by gtjamesa** [comment]
That's likely true for event/routes, but config will be caching environment variables that are set during deployment, so they can be modified post-build.

**Comment by moisish** [comment]
@dakira Personally I run event and view cache during the docker build.  The config cache cannot run during the build because the env variables are jot available.

All the options are available in the script for the maximum flexibility and can be individually disabled

**Comment by jaydrogers** [comment]
Just adding a note on this I will be adapting this PR to the new v3.0 images.

I also have a Twitter thread of related notes and discussions here: https://twitter.com/jaydrogers/status/1721574061111701630

File: src/fpm/etc/s6-overlay/scripts/laravel-automations [hunk 1]
```diff
@@ -7,6 +7,28 @@ set -e
...
 
+        ############################################################################
+        # Automated Laravel Cache
+        ############################################################################
+        if [ ${AUTORUN_LARAVEL_CONFIG_CACHE:="true"} == "true" ]; then
+            echo "🗄️ Running config cache..."
+            s6-setuidgid webuser php $WEBUSER_HOME/artisan config:cache
+        fi
+
+        if [ ${AUTORUN_LARAVEL_VIEW_CACHE:="true"} == "true" ]; then
```
Comment (a-h-abid) [inline src/fpm/etc/s6-overlay/scripts/laravel-automations:18]: I think this should be `AUTORUN_LARAVEL_EVENT_CACHE` ???

**Pull request closed without merging**